tests/unit/lib/Any_unittest.cc

```diff
@@ -836,3 +836,31 @@ TEST(Any, cast_into_sgobject)
 	EXPECT_EQ(any.cast<CSGObject*>(), ptr);
 	EXPECT_THROW(any.cast<int>(), std::logic_error);
 }
+
+TEST(Any, simple_visit)
+{
+	int extracted_value{};
+	struct ExtractVisitor
+	{
+	};
+	Any::register_visitor<int, ExtractVisitor>([&] (int value, ExtractVisitor* visitor) { extracted_value = value; });
+	auto any = make_any<int>(42);
+	any.visit<ExtractVisitor>();
+	EXPECT_EQ(any.as<int>(), extracted_value);
+}
+
+TEST(Any, stateful_visit)
+{
+	std::string value = "hello, world!";
+	struct StringStreamVisitor
+	{
+		std::stringstream ss;
+	};
+	Any::register_visitor<std::string, StringStreamVisitor>([&] (std::string value, StringStreamVisitor* visitor) {
+		(visitor->ss) << value;
+	});
+	auto any = make_any(value);
+	StringStreamVisitor visitor;
+	any.visit(&visitor);
```

Comment:
What about a couple of calls to visit to really show the stateful bit?
Perhaps I am misinterpreting the state reference.

Reply:
Well it is already stateful but I can visit twice to make it more visible